arm_robots/src/arm_robots/cartesian.py

```diff
@@ -108,7 +110,10 @@ def abort_goal(self):
         self._start_violation = 0
 
     def current_pose_in_frame(self, arm, reference_frame=None):
-        current_pose = self.motion_status_listeners[arm].get().measured_cartesian_pose
+        current_pose = None
+        # query for pose multiple times as for some reason querying once sometimes gives stale messages
+        for _ in range(RE_QUERY_ATTEMPTS):
+            current_pose = self.motion_status_listeners[arm].get().measured_cartesian_pose
```

Comment:
can you try this:
```
self.motion_status_listeners[arm].data = None
current_pose = self.motion_status_listeners[arm].get().measured_cartesian_pose
```
without the for loop. This basically clears the last data so it ensures you wait until you get a new one. I'm curious if that fixes it too.

Reply:
This does fix it too (if I block until the next message arrives); probably a better solution than requerying for an arbitrary number of times